src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -382,6 +382,11 @@ private BoundIndexerAccess BindIndexerDefaultArguments(BoundIndexerAccess indexe
         /// </summary>
         private BoundExpression CheckValue(BoundExpression expr, BindValueKind valueKind, BindingDiagnosticBag diagnostics)
         {
+            if ((valueKind & BindValueKind.Assignable) != 0)
+            {
+                expr = ResolveToExtensionMemberIfPossible(expr, diagnostics);
```

Comment:
From scenarios where we read the expression, we'll either do `BindToNaturalType` or some conversion. LHS of assignment was a remaining scenario that did hit either of those, so it needed some additional handling. 
We could do it in the caller (`BindAssignment`, etc), but this seemed a good shared location to cover more scenarios.

Reply:
Small clarification: there are some other "read" scenarios that aren't yet handled but I didn't list above. They are covered in tests with PROTOTYPE markers:
- `var x = A.B;` (no conversion since inferred, so needs special handling)
- `new[] { A.B }` (best type inference)